src/basic/chase-symlinks.c

```diff
@@ -101,6 +101,14 @@ int chase_symlinks_at(
         if (isempty(path))
                 path = ".";
 
+        if (flags & CHASE_PARENT) {
+                r = path_extract_directory(path, &buffer);
+                if (r == -EDESTADDRREQ)
+                        path = "."; /* If we don't have a parent directory, fall back to the root directory. */
```

Comment:
root directory? By that you mean "dir_fd"?

is that the right behaviour? i mean if i invoke `chase_symlinks_at(some_dir_fd, ".", CHASE_PARENT, &resolved, NULL)`, what should this result in? you say the inode some_dir_fd refers to, but that sounds surprising to me, no?

Reply:
ah, EDESTADDRREQ is not EADDRNOTAVAIL. make sense then.

but "root directory" is still misleading